articles/logging-with-winston/index.md

```diff
@@ -0,0 +1,547 @@
+- Log [formats](https://github.com/winstonjs/winston#formats). Winston provides you with several log formats. For example, when saving a log to a mongo database, the log format needs to be in JSON format. Winston solves that for you.
+- Log [profiling](https://github.com/winstonjs/winston#profiling). Winston helps you profile a block of code and measure the time the code takes to be successfully executed.
+
+### Using Winston
+
+To get started, make sure you initialize a Node.js project using `npm init`. We have a [guide](/engineering-education/beginner-guide-to-npm/) that will help you understand NPM in detail.
+
+Install the Winston package using the command `npm install winston`.
+
+Logging with Winston is simple, with just four steps as shown in the example below, and you have your a log recorded.
+
+- Add the Winston module with `require()` function.
+
+```js
+const winston = require('winston');
+```
+- Creating Winston configuration object. Winston requires at least one transport to create a log.
+
+```js
+const logConfiguration = {
+    'transports': [
+        new winston.transports.Console()
+    ]
+};
+```
+- Creating a logger and pass it to the Winston configuration object.
+
+```js
+const logger = winston.createLogger(logConfiguration);
+```
+- Log your message.
+
+```js
+// Log a message
+logger.log({
+// Message to be logged
+    message: 'Hello, Winston!',
+
+// Level of the message logging
+    level: 'info'
+});
+// Log a message
+logger.info('Hello, Winston!');
+```
+
+The above example shows you how to log to the console.
+
+```js
+{"message":"Hello, Winston logger a warning!","level":"warn"}
+{"message":"Hello, Winston logged an error!","level":"error"}
+```
+
+### Winston Transporters
+
+As a property of Winston, it supports different transports such as file transport.
+
+This will save a generated log message to a log file. The file is specified within your system. If the application creates its first log instance, the file will automatically be generated. Afterwards, any log will be saved to the created file.
+
+To do this, the logger configuration object needs to point to a file (file transporter). Simply by replacing the transport configuration object `.transports.Console()` with `.transports.File()` as shown below.
+
+```js
+transports.File({
+    filename: './logs/example.log'
+})
+```
+
+This specifies that any log generated will be saved under the log folder into the `example.log` file.
+
+The transport configuration will be:
+
+```js
+// Logger configuration
+const logConfiguration = {
+    'transports': [
+        new winston.transports.File({
+            filename: './logs/example.log'
+        })
+    ]
+};
+```
+
+Output:
+
+![Log File](/engineering-education/logging-with-winston/log-file.png)
+
```

Comment:
What's the difference between transports and multiple logging channels? Can you provide clear definitions and distinctions to readers?